### PR TITLE
Starting backend setup and first endpoint

### PR DESCRIPTION
**Pull request by manuelamercado**

### CODE REVIEW

File: backend/models.py [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+  db.create_all()
+
+'''
+User
+Entity for person that tracks exercises
+'''
+class User(db.Model):  
+  __tablename__ = 'User'
+
+  id = Column(Integer, primary_key=True)
```
Comment (jpolanco91) [inline backend/models.py]: TODO: Use hash for this key. If not hash type found save the hash as string or char

File: backend/config.py [hunk 2]
```diff
@@ -0,0 +1,11 @@
+import os
+SECRET_KEY = os.urandom(32)
```
Comment (jpolanco91) [inline backend/config.py]: Let's check this to see if it's used

File: backend/api.py [hunk 3]
```diff
@@ -0,0 +1,129 @@
...
+  # create and configure the app
+  app = Flask(__name__)
+  setup_db(app)
+  migrate = Migrate(app, db)
+  CORS(app, resources={r"/*": {"origins": "*"}})
+
+  # CORS Headers
+  @app.after_request
+  def after_request(response):
+    response.headers.add('Access-Control-Allow-Headers', 'Content-Type,Authorization,true')
```
Comment (jpolanco91) [inline backend/api.py]: What's the `true` value for in `Access-Control-Allow-Headers`

**Review by jpolanco91 — CHANGES_REQUESTED** [review]
-Comment the unused `SECRET_KEY` config.
-Verify why do we have a `true` value in `Access-Control-Allow-Headers`
-Investigate about PEP8 and make the code compliant with it.
-Complete the GET endpoint for users (using log and userId query params).
-Make the GET/POST endpoints for exercises.

File: backend/models.py (hunk 1 above)
Comment (manuelamercado) [reply backend/models.py]: Postgres has a type called UUID. The only difference is that it adds some `-`.

File: backend/config.py (hunk 2 above)
Comment (manuelamercado) [reply backend/config.py]: This is for signing the session cookies. In this way the secret is reset every time the app start, so all the users will lose the sessions if the server is restarted.

File: backend/api.py (hunk 3 above)
Comment (manuelamercado) [reply backend/api.py]: This is the value to accept the `Access-Control-Allow-Headers`

File: backend/app.py [hunk 4]
```diff
@@ -58,13 +58,138 @@ def add_user():
...
+
+        if len(users_data):
+            return jsonify({
+              'users': users
+            })
+        else:
+            abort(404)
+
+        '''
+    POST /api/exercise/add
```
Comment (jpolanco91) [inline backend/app.py]: Quoted comments should be starting at the first line of the function

File: backend/app.py [hunk 5]
```diff
@@ -58,13 +58,138 @@ def add_user():
...
+                  'date': exercise.exercise_date
+                })
+
+        except Exception as e:
+            print(sys.exc_info(), e)
+            abort(422)
+
+    '''
+    GET /api/exercise/log
+    returns a json {'count': count} where count is the number of exercises
```
Comment (jpolanco91) [inline backend/app.py]: Same here. Move this to the first line of the function's body.

File: backend/app.py [hunk 6]
```diff
@@ -58,13 +58,138 @@ def add_user():
...
+                if from_date:
+                    exercises_data = Exercise.query.filter(
+                      Exercise.user_uuid == user_id,
+                      Exercise.exercise_date >= from_date
+                    ).all()
+                    if to_date:
+                        exercises_data = Exercise.query.filter(
+                          Exercise.exercise_date <= to_date,
+                          Exercise.exercise_date >= from_date,
+                          Exercise.exercise_date <= to_date
```
Comment (jpolanco91) [inline backend/app.py:158]: Duplicated line of code

File: backend/app.py [hunk 7]
```diff
@@ -58,13 +58,138 @@ def add_user():
...
+                if to_date:
+                    exercises_data = Exercise.query.filter(
+                      Exercise.user_uuid == user_id,
+                      Exercise.exercise_date <= to_date
+                    ).all()
+                    if from_date:
+                        exercises_data = Exercise.query.filter(
+                          Exercise.exercise_date <= to_date,
+                          Exercise.exercise_date >= from_date,
+                          Exercise.exercise_date <= to_date
```
Comment (jpolanco91) [inline backend/app.py]: Same as before, duplicated line of code

File: backend/app.py [hunk 8]
```diff
@@ -58,13 +58,138 @@ def add_user():
...
+                        ).all()
+
+                exercises = [exercise.short() for exercise in exercises_data]
+                if limit:
+                    exercises_data = exercises[0:limit]
+
+
+                if len(exercises_data):
```
Comment (jpolanco91) [inline backend/app.py]: Should check if it's not equal to 0 using `!=`

File: backend/app.py [hunk 9]
```diff
@@ -58,13 +58,138 @@ def add_user():
...
+                    return jsonify({
+                      'user': user,
+                      'log': [exercises]
+                    })
+                else:
+                    abort(404)
```
Comment (jpolanco91) [inline backend/app.py]: Should return an empty object. 404 errors are exclusive for when an endpoint/route/page does not exist.

**Review by jpolanco91 — CHANGES_REQUESTED** [review]
Looking good so far, only to fix some little styling and code feedbacks

**Comment by jpolanco91** [comment]
Also for the next session, investigate how to simplify the the `from_date` and `to_date` logic.

**Review by jpolanco91 — DISMISSED** [review]
LGTM !! merge it !!

**Pull request merged**